pkg/util/timeutil/pgdate/field_extract.go

```diff
@@ -765,7 +765,7 @@ func (fe *fieldExtract) Reset(field field, v int) error {
 // the field has already been set.
 func (fe *fieldExtract) Set(field field, v int) error {
 	if !fe.wanted.Has(field) {
-		return errors.AssertionFailedf("field %s is not wanted in %v", errors.Safe(field.Pretty()), errors.Safe(fe.wanted))
+		return inputErrorf("field %s is not wanted in %v", errors.Safe(field.Pretty()), errors.Safe(fe.wanted))
```

Comment:
Can you avoid errors.Safe here

1) in the user-provided value, which is definitely unsafe

2) for the field name, instead have the field implement a SafeFormat method, or have a SafePretty method that returns redact.RedactableString